### ISSUE TITLE
Easy way to run same script on multiple connections?

### ISSUE THREAD

**Issue by shlomiw** [dbeaver/dbeaver#3605]
Hi,
Is it possible to execute same script on multiple different DB connections at once?
i.e.
I have a script and 10 different DB connections (same schema).
Now I have to open SQL script editor tab for each connection, copy+paste the script and execute it.

Any ideas how to expedite this process inside DBeaver?
Thanks

**Comment by p91paul** [comment]
I usually just change the connection on the top toolbar without creating a new script, this should save you a few clicks.
However the ability to simultaneously execute the same sql on many connections is something I could use too!

**Comment by shlomiw** [comment]
Hi @p91paul , tnx for your reply!
The problem is that my scripts are sometimes run for a long time.. so I'd like to execute it in parallel on all DB's.

**Comment by serge-rider** [comment]
This makes sense. I think we'll need new UI for that (some special command like "Execute in multiple connections").

**Comment by serge-rider** [comment]
After all I think making just special command isn't good  enough.
Its better to make new "composite" connection. It will contain from multiple other existing connections. Each query executed in this connection will be executed in every "nested" connection.
Basically this is not very hard to do while we do not want to perform cross-database joins and stuff like that.

One of questions is how to represent query results. Say we have composite connection which consists from 3 other connections. E.g. you execute "SELECT * FROM TABLE".
Should it produce 3 tabs for query results? How to visualize this (how to determine which tab belongs to which connection). 

Another question - should it execute all queries in separate threads or execute them one by one?

**Comment by shlomiw** [comment]
Hi, as I see it, simply run the same script in separated tabs in parallel. This what I'm doing now manually.
What do you think?

**Comment by percyvega** [comment]
If I could just duplicate the script view, so that I point one view of the script to STG and another view of the same script to PRD, that would solve my issues. This would be similar to what IntelliJ has: "Split Vertically" and "Move Right".

**Comment by serge-rider** [comment]
@percyvega Regarding UI - you can do the same in DBeaver (main menu Window->Editor->Split editor horizontal/vertical). 
Of course it is not related to multi-connections but you can open two editors with PROD/STG connections and see them on screen at once.

**Comment by rmarkus** [comment]
Yes please provide an option to have multi-DB connections. 
The usecase for me is: 
- database with Read-only access (Oracle)
- csv-Data source (or SQLite source) read-write
- requirement to process both, ie for each entry in cvs grab some data from Oracle and write back to a different csv

**Comment by p91paul** [comment]
@rmarkus this ticket is about running the same sql on different connections, not combining data from different sources. What you request would make DBeaver an ETL tool, which is an entirely different software category. You might want to open a new issue with your request, maybe it will be considered.

**Comment by MouseOfTheDesert** [comment]
Hi, I have the exact same need as shlomiw , and i was hoping this feature to be released on 6.0 . Please keep this issue opened

**Comment by ChaminW** [comment]
Hi, I have the same requirement. Have to execute the same query for multiple slave DB clusters which take significant time.

**Comment by ChaminW** [comment]
What I suggest is having dynamic data sources for scripts rather having a single active data source. Then we can right query like 
SELECT * FROM <DATA_SOURCE>.<SCHEMA>.<TABLE>
where <SOME_CONDITION>;

**Comment by dicaetano** [comment]
Please, I have more then 30 connections, and sometimes I have to run a query in all of then. I already love DBeaver, but this would take it to another level.

**Comment by gabrielbet** [comment]
@dicaetano > "Please, I have more then 30 connections, and sometimes I have to run a query in all of then. I already love DBeaver, but this would take it to another level."

I share the same felling 🙏

**Comment by rat** [comment]
Hi, I have the same requirement. I usually need to do an inset or update on a postgres server that has 100 or 200 databases, everyone needs to receive the update or insert

I usually need to open database by database and run, it takes a whole day. the problem is that it needs to be done outside business hours, so that customers are not left without a system.

**Comment by serge-rider** [comment]
New SQL script task will support any number of scripts and any number of connections.
Each script will be executed in all selected connections.

Task view->Create New Task->SQL Script.

**Comment by serge-rider** [comment]
![image](https://user-images.githubusercontent.com/6398845/75249167-bccf0580-57e6-11ea-97e6-9e0d6acbf5db.png)

**Comment by uslss** [comment]
Reopened.
1. Connection could be added more than one time.
2. Scripts folder could be added instead of script, but scripts from folder aren't added
![image](https://user-images.githubusercontent.com/31996417/75268710-ea2cab00-5808-11ea-9742-9a6b09b0ee49.png)
3. Checking/unchecking auto-commit option doesn't have any effect.
 Case: 
    I've got two update script. The second script is incorrect and causes error. Table from first script is always updated (no matter if auto-commit disabled or enabled)
4. "Dump query results to log file" option also have got no effect. Log looks the same
![image](https://user-images.githubusercontent.com/31996417/75269150-822a9480-5809-11ea-8fb5-fa607adf87f9.png)

**Comment by serge-rider** [comment]
Fixed

**Comment by uslss** [comment]
Verified.
Related issue #8017

**Comment by shlomiw** [comment]
@serge-rider - thank you very much! appreciated

**Comment by jason1004** [comment]
Is it possible to execute a SQL script in multiple databases in one connection?

**Comment by avandorp** [comment]
Created task #16331  to make logging the query results of the script possible. Very straightforward implementation to solve my personal issue, but debatably better than having a UI presented to the user which never does anything.

**Comment by arktechltd** [comment]
I have a select statement that I want to run on X connections, but I want to see the results. How to do that using Tasks? Now when I run this in a task, I simply see this in logs:

`2024-04-02 02:17:51.014 - -----------------> Execute query
SELECT first_name, user_type_id
	FROM public.users
2024-04-02 02:17:51.269 - 	Fetch result set
2024-04-02 02:17:51.269 - 	14 rows fetched
2024-04-02 02:17:51.270 - -----------------> Time: 0.255s, fetched 14 row(s)`

I need to see those 14 rows (considering that I have 100 connections with same logs above)

Thanks.

**Comment by avandorp** [comment]
> I have a select statement that I want to run on X connections, but I want to see the results. How to do that using Tasks? Now when I run this in a task, I simply see this in logs:
> 
> `2024-04-02 02:17:51.014 - -----------------> Execute query SELECT first_name, user_type_id FROM public.users 2024-04-02 02:17:51.269 - Fetch result set 2024-04-02 02:17:51.269 - 14 rows fetched 2024-04-02 02:17:51.270 - -----------------> Time: 0.255s, fetched 14 row(s)`
> 
> I need to see those 14 rows (considering that I have 100 connections with same logs above)
> 
> Thanks.

Did you check the checkmark "Dump query results to log file"?

**Comment by arktechltd** [comment]
> > I have a select statement that I want to run on X connections, but I want to see the results. How to do that using Tasks? Now when I run this in a task, I simply see this in logs:
> > `2024-04-02 02:17:51.014 - -----------------> Execute query SELECT first_name, user_type_id FROM public.users 2024-04-02 02:17:51.269 - Fetch result set 2024-04-02 02:17:51.269 - 14 rows fetched 2024-04-02 02:17:51.270 - -----------------> Time: 0.255s, fetched 14 row(s)`
> > I need to see those 14 rows (considering that I have 100 connections with same logs above)
> > Thanks.
> 
> Did you check the checkmark "Dump query results to log file"?

@avandorp I couldn't find this check. Can you please guide me where? I looked for this in the task execution page but it's not there. I'm having postgres connections.

Thanks

**Comment by avandorp** [comment]
> > Did you check the checkmark "Dump query results to log file"?
> 
> @avandorp I couldn't find this check. Can you please guide me where? I looked for this in the task execution page but it's not there. I'm having postgres connections.
> 
> Thanks

![grafik](https://github.com/dbeaver/dbeaver/assets/18458913/55c0d116-c72e-4bd9-85ce-323b3a867147)

**Comment by arktechltd** [comment]
> > > Did you check the checkmark "Dump query results to log file"?
> > 
> > 
> > @avandorp I couldn't find this check. Can you please guide me where? I looked for this in the task execution page but it's not there. I'm having postgres connections.
> > Thanks
> 
> ![grafik](https://private-user-images.githubusercontent.com/18458913/318742103-55c0d116-c72e-4bd9-85ce-323b3a867147.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwNTM4ODgsIm5iZiI6MTcxMjA1MzU4OCwicGF0aCI6Ii8xODQ1ODkxMy8zMTg3NDIxMDMtNTVjMGQxMTYtYzcyZS00YmQ5LTg1Y2UtMzIzYjNhODY3MTQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAyVDEwMjYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5NGYxMjYzZDU1N2U4NDkzMmE1N2I3OGU1ZGVkOGQ1ZWU3YWYwNDM5M2Y4MDNkNzllNmFlZjdmNjc5OThhODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v1sNWuwOxTgiON08ODl0bV3OY_A7gYczWhw1WBiMmEc)

Thanks alot @avandorp . I had to create a new task to see this option, It was not there when I tried to update and never noticed this before while creating the task. This did the job.
